### PR TITLE
Created commands

### PR DESCRIPTION
**Pull request by kristoferssolo**
Created 3 vim commands for user to run:
1. `:Runner` - wraps `require('runner').run()`;
2. `:AutoRunner` - wraps `require('runner').autorun()`;
3. `:AutoRunnerStop` - wraps `require('runner').autorun_stop()`;

The `AutoRuner` command automatically runs the code every time any file is saved.
The `AutoRunerStop` command stops `AutoRunner` and closes the window.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,8 +54,20 @@ A customizable Neovim plugin to run code inside the editor
 
 ## Usage
 
+### Commands
+
+| Command           | Description                              |
```
Comment (MarcHamamji) [inline README.md]: Add a table column to show which function each command runs, so it's more clear.

File: README.md [hunk 2]
```diff
@@ -53,8 +54,20 @@ A customizable Neovim plugin to run code inside the editor
...
+
+### Lua
+
   ```lua
   require('runner').run()
+  require('runner').autorun()
+  require('runner').autorun_stop()
 
   -- Can also be called with the buffer number where the code is:
   -- require('runner').run( <buffer_number> )
```
Comment (MarcHamamji) [inline README.md:73]: Specify that both functions `run` and `autorun` can accept the buffer number as an argument.

File: lua/runner/init.lua [hunk 3]
```diff
@@ -28,29 +28,44 @@ M.set_handler = function(filetype, handler)
...
+end
+
+--- @param bufnr integer|nil
+M.autorun = function(bufnr)
+  M.run(bufnr)
+  vim.api.nvim_create_autocmd('BufWritePost', {
+    group = vim.api.nvim_create_augroup('AutoRunner', { clear = true }),
+    pattern = '*',
+    callback = function()
+      helpers.shell_handler(utils._last_command, false)()
```
Comment (MarcHamamji) [inline lua/runner/init.lua]: You're restricting the command to only be a shell command. That way, if for example you're running a lua file, which uses the command handler to run `luafile %`, it breaks. Please update it to run the last handler regardless of which type it is.
You can add a `_last_handler` function somewhere that you overwrite each time you run the file, that just encapsulates the handler being run. When `autorun` runs, it just calls that function.

**Review by MarcHamamji — CHANGES_REQUESTED** [review]
Thank you very much for your work! I added some comments for things I would like you to update. :)

**Review by MarcHamamji — APPROVED** [review]

**Comment by MarcHamamji** [comment]
Great, I just have one more question: Is there a reason you create the user commands in the `plugin` directory? Why not in the `setup()`?

**Comment by kristoferssolo** [comment]
I followed examples of other plugins and [this template](https://github.com/m00qek/plugin-template.nvim). Everyone inserts code that needs to be executed when the plugin is loaded into the `plugins` directory, e.g. `user_commands`. Some even put `setup()` in there.

**Comment by MarcHamamji** [comment]
Reading [this](https://www.reddit.com/r/neovim/comments/y58zh8/comment/isio0xk/?utm_source=share&utm_medium=web2x&context=3), I think that it's better to just put everything in the setup, so that the commands are only created when the `setup` function is called.

**Review by MarcHamamji — CHANGES_REQUESTED** [review]
Reading [this](https://www.reddit.com/r/neovim/comments/y58zh8/comment/isio0xk/?utm_source=share&utm_medium=web2x&context=3), I think that it's better to just put everything in the setup, so that the commands are only created when the `setup` function is called. So can you please just copy the contents of that file and put them in the setup? Also can you modify the README to say that the setup is not optional anymore?

**Comment by kristoferssolo** [comment]
Hopefully I understood you correctly

**Review by MarcHamamji — APPROVED** [review]

**Comment by MarcHamamji** [comment]
Perfect, thanks a lot!

**Pull request merged**